ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/AuthenticationPluginHandler.java

```diff
@@ -40,25 +43,38 @@
 
   private final Server server;
   private final AuthenticationPlugin securityHandlerPlugin;
-  private final Set<String> unauthenticatedPaths = new HashSet<>();
+
+  private final Pattern unauthedPathsPattern;
 
   public AuthenticationPluginHandler(final Server server,
       final AuthenticationPlugin securityHandlerPlugin) {
     this.server = Objects.requireNonNull(server);
     this.securityHandlerPlugin = Objects.requireNonNull(securityHandlerPlugin);
     // We add in all the paths that don't require authorization from
     // KsqlAuthorizationProviderHandler
-    unauthenticatedPaths.addAll(KsqlAuthorizationProviderHandler.PATHS_WITHOUT_AUTHORIZATION);
+    final Set<String> unauthenticatedPaths = new HashSet<>(
+        KsqlAuthorizationProviderHandler.PATHS_WITHOUT_AUTHORIZATION);
     // And then we add anything from the property authentication.skip.paths
     // This preserves the behaviour from the previous Jetty based implementation
     final List<String> unauthed = server.getConfig()
         .getList(ApiServerConfig.AUTHENTICATION_SKIP_PATHS_CONFIG);
     unauthenticatedPaths.addAll(unauthed);
+    final StringBuilder pathBuilder = new StringBuilder();
```

Comment:
Replace this with `String.join(",", unauthenticatedPaths)`?

Reply:
I missed have missed this method when it came out in Java 8, nice pointer :)